src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc

```diff
@@ -418,9 +408,13 @@ tsi_result alts_tsi_handshaker_create(
   handshaker->has_created_handshaker_client = false;
   handshaker->handshaker_service_url = gpr_strdup(handshaker_service_url);
   handshaker->options = grpc_alts_credentials_options_copy(options);
-  handshaker->base.vtable = handshaker->use_dedicated_cq
-                                ? &handshaker_vtable_dedicated
-                                : &handshaker_vtable;
+  handshaker->base.vtable =
+      use_dedicated_cq ? &handshaker_vtable_dedicated : &handshaker_vtable;
+  handshaker->channel =
+      use_dedicated_cq
+          ? nullptr
+          : grpc_insecure_channel_create(handshaker->handshaker_service_url,
```

Comment:
use of this API has <i>potentially</i> (though maybe not in practice) the same issue as the one that was fixed by https://github.com/grpc/grpc/pull/17148 - using a core surface API and initting `ExecCtx` within core?